external-crates/move/tools/move-package/src/resolution/dependency_graph.rs

```diff
@@ -140,84 +126,706 @@ struct PackageWithResolverTOML<'a>(&'a Package);
+        }
+
+        for to_pkg_name in self
+            .package_graph
+            .neighbors_directed(from_pkg_name, Direction::Outgoing)
+        {
+            self.find_pruned_pkgs(
+                pruned_pkgs,
+                reachable_pkgs,
+                root_pkg_name,
+                to_pkg_name,
+                mode,
+                overrides,
+                dev_overrides,
+                override_found,
+            )?;
+        }
+        Ok(())
+    }
+
+    /// Prunes packages in a sub-graph based on the overrides information from the outer graph.
+    fn prune_overriden_pkgs(
+        &mut self,
+        root_pkg_name: PM::PackageName,
+        from_pkg_name: PM::PackageName,
+        mode: DependencyMode,
+        overrides: &BTreeMap<PM::PackageName, Package>,
+        dev_overrides: &BTreeMap<PM::PackageName, Package>,
+    ) -> Result<()> {
+        let mut pruned_pkgs = BTreeMap::new();
+        let mut reachable_pkgs = BTreeSet::new();
+        self.find_pruned_pkgs(
+            &mut pruned_pkgs,
+            &mut reachable_pkgs,
+            root_pkg_name,
+            from_pkg_name,
+            mode,
+            overrides,
+            dev_overrides,
+            false,
+        )?;
+
+        for (pkg, neighbors) in pruned_pkgs {
+            // simply removing a node does not remove edges coming from its neighbors to this node
+            for n in neighbors {
+                self.package_graph.remove_edge(n, pkg);
+            }
+            self.package_graph.remove_node(pkg);
+            self.package_table.remove(&pkg);
+        }
+
+        Ok(())
+    }
+
+    /// Given all sub-graphs representing dependencies of the parent manifest file, combines all
+    /// subgraphs to form the parent dependency graph.
+    pub fn merge(
+        &mut self,
+        dep_graphs: BTreeMap<PM::PackageName, (DependencyGraph, bool)>,
+        mode: DependencyMode,
+        root_package: PM::PackageName,
+        parent: &PM::DependencyKind,
+        dependencies: &PM::Dependencies,
+    ) -> Result<()> {
+        if !self.always_deps.is_empty() {
+            bail!("Merging dependencies into a graph after calculating its 'always' dependencies");
+        }
+
+        // collect all package names in all graphs in package table
+        let mut all_packges: BTreeSet<PM::PackageName> = BTreeSet::new();
+        for (g, _) in (&dep_graphs).values() {
+            all_packges.extend(g.package_table.keys());
+        }
+
+        // analyze all packages in all package tables of all sub-graphs to determine if any of these
+        // packages represent a conflicting dependency; insert the packages and their respective
+        // edges into the combined graph along the way
+        for pkg_name in all_packges {
+            let mut existing_pkg_info: Option<(Symbol, &DependencyGraph, &Package)> = None;
+            for (dep_name, (g, _)) in &dep_graphs {
+                if let Some(pkg) = g.package_table.get(&pkg_name) {
```

Comment:
Can you include `self` among the `dep_graphs` and include the immediate dependencies among `all_packages` to avoid having to special case them?

Reply:
Sorry for some reason when I reviewed this last, I didn't see your reply!  Not sure what happened there, but yes, what I was suggesting was for the self graph to be pre-populated with the edges to its immediate dependencies, and then subsequently included in the merge.